src/Microsoft.Framework.Logging.TraceSource/TraceSourceLoggerProvider.cs

```diff
@@ -97,5 +99,26 @@ private static bool HasDefaultSwitch(DiagnosticsTraceSource traceSource)
             return string.IsNullOrEmpty(traceSource.Switch.DisplayName) == string.IsNullOrEmpty(traceSource.Name) &&
                 traceSource.Switch.Level == SourceLevels.Off;
         }
+
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    _rootTraceListener.Flush();
+                    _rootTraceListener.Dispose();
+                }
+
+                _disposed = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
Note that the entire msdn write-up is discussing native resources, safe handles, finalizers, etc.. We're not doing any of that, nor do we expect anyone to derive from these types and do it.  We don't need any more than:
```c#
public void Dispose()
{
  if (!disposed)
  {
    _disposed = true;
    _rootTraceListener.Flush();
    _rootTraceListener.Dispose();
  }
}
```
That said, I don't recommend marking it sealed either, there's no reason to forbid someone from deriving from it.